### ISSUE TITLE
line segmenter's DataProvider should have some line break property

### ISSUE THREAD

**Issue by makotokato** [unicode-org/icu4x#1637]
Actually line segmenter code references AI, AL, BA, CJ, CM, EX, H2, H3, HY, ID, IN, JL, JT, JV, NS, NU, PO_EAW, PR_EAW, SA and XX from generated state machine table directly, I think we should move this to `DataProvider`/`LineBreakDataV1`.  Word segmenter (rule segmenter) has  some property index such as SOT into `RuleBreakDataV1`.

CC: @aethanyc

**Comment by aethanyc** [comment]
Yup, I also think about this problem recently. These symbol should be defined in `LineBreakDataV1`.

The numeric value of these symbols AI, AL, ... are tight to the data in `line.toml`, and can be referenced by utilities like https://github.com/unicode-org/icu4x/blob/4c22ef7d6aa78aa8d1bc4f7193619ccd03f65976/experimental/segmenter/src/line.rs#L319-L343

The most straightforward way is to add more fields in
```
pub struct LineBreakDataV1 {
  ...
  pub ai_property: u8,
  pub al_property: u8,
  ...
}
```

But, this soon become tedious. I'm not sure if there's a better way.

**Comment by aethanyc** [comment]
Maybe we can store the numeric values in `line.toml` so that it can decouple from the numeric value in [`impl LineBreak`]( https://github.com/unicode-org/icu4x/blob/92a15945dbea37e761e9c8be1eab1edc41f05f9c/components/properties./src/props.rs#L535-L579)

```
[[tables]]
name = "AI"
value = 1

[[tables]]
name = "AL"
value = 2
```

**Comment by makotokato** [comment]
Since this is hard-coded, when updating line.toml for newer spec, we have to increment version of `LineBreakData` etc.
